synapse/cortex.py

```diff
@@ -4742,8 +4742,8 @@ async def addStormDmon(self, ddef):
         '''
         Add a storm dmon task.
         '''
-        iden = s_common.guid()
-        ddef['iden'] = iden
+        if ddef.get('iden') is None:
+            ddef['iden'] = s_common.guid()
```

Comment:
We should raise a s_exc.DupIden exception in the top handler here, similar to how `addCronJob(...)` works.